plugins/inputs/intel_powerstat/intel_powerstat.go

```diff
@@ -97,7 +100,17 @@ func (p *PowerStat) Gather(acc telegraf.Accumulator) error {
 	}
 
 	if p.areCoreMetricsEnabled() {
-		p.addPerCoreMetrics(acc)
+		if p.msr.isMsrLoaded() {
+			p.logOnce["msr"] = nil
+			p.addPerCoreMetrics(acc)
```

Comment:
Now you are checking if the MSR is loaded, was this a bug from the last PR?

Reply:
Checking msr here it allows us to throw less logs / errors.